### PR TITLE
Data export

### PR DESCRIPTION
**Pull request by ayeswarya**
Resolves #

## Description

Add urlForDownloadQuery function which will be used for async data export implementation
## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/services/navi-facts.ts [hunk 1]
```diff
@@ -60,6 +60,18 @@ export default class NaviFactsService extends Service {
...
   }
 
+  /**
+   * @method getURL - Uses the adapter to get the download query url for the request
+   * @param {Object} request - request object
+   * @param {Object} [options] - options object
+   * @returns {String} - url for the request
+   */
+  getDownloadURL(request: RequestV2, options: RequestOptions) {
+    const type = config.navi.dataSources[0].type;
```
Comment (jkusa) [inline packages/data/addon/services/navi-facts.ts:70]: Can we use the request to get the dataSourceName then lookup the type

File: packages/data/tests/unit/adapters/facts/bard-test.ts [hunk 2]
```diff
@@ -1077,6 +1077,107 @@ module('Unit | Adapter | facts/bard', function(hooks) {
     );
   });
 
+  test('urlForDownloadQuery', async function(assert) {
+    assert.expect(7);
+
+    assert.rejects(
+      //@ts-expect-error
```
Comment (jkusa) [inline packages/data/tests/unit/adapters/facts/bard-test.ts:1084]: Why is this needed?

Comment (kevinhinterlong) [reply packages/data/tests/unit/adapters/facts/bard-test.ts:1084]: It's testing that it throws an error if we pass in a v1 request. TS knows that this should be `requestVersion: '2.0'` and complains but we're expecting it

Comment (jkusa) [reply packages/data/tests/unit/adapters/facts/bard-test.ts:1084]: Ah, yeah the interface supports it but not the implementation. I say we are V2 all the way and remove this test.

Comment (ayeswarya) [reply packages/data/tests/unit/adapters/facts/bard-test.ts:1084]: Removed said test

File: packages/data/addon/services/navi-facts.ts [hunk 3]
```diff
@@ -67,7 +67,9 @@ export default class NaviFactsService extends Service {
    * @returns {String} - url for the request
    */
   getDownloadURL(request: RequestV2, options: RequestOptions) {
-    const type = config.navi.dataSources[0].type;
+    const type = config.navi.dataSources.filter(dataSources => {
+      return dataSources.name === request.dataSource;
+    })[0].type;
```
Comment (kevinhinterlong) [inline packages/data/addon/services/navi-facts.ts:72]: Something like this
https://github.com/yahoo/navi/blob/641a570c54f8b9990094b32442bd42b301578816/packages/data/addon/services/navi-dimension.ts#L9
https://github.com/yahoo/navi/blob/641a570c54f8b9990094b32442bd42b301578816/packages/data/addon/services/navi-dimension.ts#L42-L43

File: packages/data/tests/unit/adapters/facts/bard-test.ts [hunk 4]
```diff
@@ -1077,18 +1068,102 @@ module('Unit | Adapter | facts/bard', function(hooks) {
...
+          values: ['2015-01-03', '2015-01-04']
+        }
+      ],
+      sorts: [
+        {
+          type: 'metric',
+          field: 'm1',
+          parameters: {},
+          //@ts-expect-error
+          direction: undefined
```
Comment (mattgraham1995) [inline packages/data/tests/unit/adapters/facts/bard-test.ts:1126]: why is direction undefined?

File: packages/data/tests/unit/adapters/facts/bard-test.ts [hunk 5]
```diff
@@ -1077,18 +1068,102 @@ module('Unit | Adapter | facts/bard', function(hooks) {
     );
   });
 
-  test('fetchDataForRequest', function(assert) {
-    assert.expect(2);
+  test('urlForDownloadQuery', async function(assert) {
+    assert.expect(6);
+    assert.equal(
+      decodeURIComponent(await Adapter.urlForDownloadQuery(TestRequest)),
+      HOST +
```
Comment (mattgraham1995) [inline packages/data/tests/unit/adapters/facts/bard-test.ts:1075]: Could we use template literals for the expected strings?

e.g.
````
`${HOST}/v1/data/table1/grain1/d1/d2/?dateTime=2015-01-03/2015-01-04&metrics=m1,m2,r(p=123)&filters=d3|id-in["v1","v2"],d4|id-in["v3","v4"],d5|id-notin[""]&having=m1-gt[0]&format=json`
````

**Review by mattgraham1995 — APPROVED** [review]

File: packages/data/tests/unit/adapters/facts/bard-test.ts [hunk 6]
```diff
@@ -1038,57 +1021,115 @@ module('Unit | Adapter | facts/bard', function(hooks) {
...
+          type: 'timeDimension',
+          operator: 'bet',
+          values: ['2015-01-03', '2015-01-04']
+        }
+      ]
+    };
+    assert.equal(
+      decodeURIComponent(await Adapter.urlForDownloadQuery(onlyDateFilter)),
+      `${HOST}/v1/data/table1/grain1/d1/d2/?dateTime=2015-01-03/2015-01-04&metrics=m1,m2,r(p=123)&format=json`,
+      'urlForDownloadQuery correctly built the URL for a request with no filters'
```
Comment (kevinhinterlong) [inline packages/data/tests/unit/adapters/facts/bard-test.ts:1082]:
```suggestion
      'urlForDownloadQuery correctly built the URL for a request with only date filter'
```

**Review by kevinhinterlong — APPROVED** [review]

**Review by jkusa — APPROVED** [review]
